lib/api_analytics.js

```diff
@@ -0,0 +1,74 @@
+'use strict'
+
+
+// const _ = require('lodash');
+
+let tally = {}
+
+
+/*
+function _avgTime (avgArray) {
+    const total = avgArray.reduce((total, time) => {
+        return total + time;
+    });
+    return total/avgArray.length;
+}
+
+
+function _flush() {
+    // resets the tally
+    console.log(tally);
+
+    _.forIn(tally, (i) => {
+            const avgSearchTime = _avgTime(i.searchTimes);
+            delete i.searchTimes;
+            i.avgSearchTime = avgSearchTime;
+    });
+    console.log('adding to es');
+    console.log('resetting tally');
+
+    console.log(tally);
+    tally = {};
+}
+*/
+
+// setInterval(flush, 10000);
+
+function tokenAndStatusCodeCount(req, res, searchTime) {
+    console.log(req.user.api_token);
+    const apiToken = req.user.api_token.slice(0,5);
+    const statusCode = res.statusCode;
+    const pathname = req._parsedOriginalUrl.pathname;
+
+    const pathArray = pathname.split("/");
+    const apiEndpoint = pathArray[pathArray.length - 1];
```

Comment:
I think we probably want to keep the full path here in case the same endpoint is registered under different bases. So `/api/v1/endpoint` vs. `/api/v2/endpoint`

Reply:
removed those lines